polyfill/lib/ecmascript.mjs

```diff
@@ -204,20 +204,21 @@ export const ES = ObjectAssign({}, ES2019, {
   ParseTemporalDurationString: (isoString) => {
     const match = PARSE.duration.exec(isoString);
     if (!match) throw new RangeError(`invalid duration: ${isoString}`);
-    if (match.slice(1).every((element) => element === undefined)) {
+    if (match.slice(2).every((element) => element === undefined)) {
       throw new RangeError(`invalid duration: ${isoString}`);
     }
-    const years = ES.ToInteger(match[1]);
-    const months = ES.ToInteger(match[2]);
-    const weeks = ES.ToInteger(match[3]);
-    const days = ES.ToInteger(match[4]);
-    const hours = ES.ToInteger(match[5]);
-    const minutes = ES.ToInteger(match[6]);
-    const seconds = ES.ToInteger(match[7]);
-    const fraction = match[8] + '000000000';
-    const milliseconds = ES.ToInteger(fraction.slice(0, 3));
-    const microseconds = ES.ToInteger(fraction.slice(3, 6));
-    const nanoseconds = ES.ToInteger(fraction.slice(6, 9));
+    const sign = match[1] === '-' ? -1 : 1;
```

Comment:
ISO 8601 for offsets accepts both an ASCII hyphen `-` and also a unicode minus sign `−`.  Should we accept both here too?  Here's an excerpt from Wikipedia:

> To represent a negative offset, ISO 8601 specifies using either a hyphen–minus or a minus sign character. If the interchange character set is limited and does not have a minus sign character, then the hyphen–minus should be used. ASCII does not have a minus sign, so its hyphen–minus character (code is 45 decimal or 2D hexadecimal) would be used. If the character set has a minus sign, then that character should be used. Unicode has a minus sign, and its character code is U+2212 (2212 hexadecimal); the HTML character entity invocation is &minus;.

Reply:
FWIW, ISO offset parsing also doesn't accept U+2212, so if we fix it here we should probably fix offset parsing at the same time.  Also, the `parseOffsetString` local function is repeated in both ecmascript.mjs and absolute.mjs. Should it be?

Reply:
I agree that it could be supported but I'd rather open a separate issue for it.

> the `parseOffsetString` local function is repeated in both ecmascript.mjs and absolute.mjs

There was a reason for that, but I can't remember it right now.

Reply:
#814

Reply:
Sounds good, thanks.